### DISCUSSION TITLE
Is there anyway to check what version of the ZMK firmware is currently installed?

### DISCUSSION THREAD

**Discussion by ecerulm** [KinesisCorporation/Adv360-Pro-ZMK#136] [Q&A] [answered]
Well, I know that it's 2.0 since I just build and flashed it. But if there anyway to get a confirmation from the keyboard itself? 

When I run `lsusb`  I get

```
Bus 020 Device 024: ID 29ea:0362 29ea Adv360 Pro  Serial: C010xxxxxxxxx
```

on the macOS System Information it appears as 

```
Product ID: 0x0362
Vendor ID: 0x29ea
Version: 3.00
Serial Number: C010xxxx 
Manufacturer: Kinesis Corporation
```

I don't think the `Version 3.00` is the firmware version. 

It's the firmware version exposed anywhere?

**Comment by ReFil** [comment]
No, not easily, especially given that ZMK doesn't have strict versioning.

**Comment by ReFil** [comment, marked as answer]
We now have a versioning system implemented: https://github.com/KinesisCorporation/Adv360-Pro-ZMK/pull/267

**Reply by ecerulm** [reply]
I just built and installed the V3.0  (commit cdcec22e609688812bf0f5ac303cc9c6970efb23)

Pressing **Mod+V** types "20231116-3.0-22"

So if pressing Mod+V produces nothing that means that the keyboard is running on a previous version.

**Reply by ReFil** [reply]
Yes, this only starts versioning new firmwares

**Reply by pthieu** [reply]
i only have access to zmk.studio, is there a way to set a keybind to do `MOD + V`?

**Reply by ReFil** [reply]
The default firmware should do that
